### ISSUE TITLE
Проблемы с кнопкой удаления коммента

### ISSUE THREAD

**Issue by Farxial** [everypony/tabun#149]
- [ ] Не работает (в веб-консоли "TypeError: this is not a function")
- [ ] Появляется только при нажатии на кнопку изменения

**Comment by zluca** [comment]
> Появляется только при нажатии на кнопку изменения

Дык я так и задумывал. =) Но если хош переделать смотри comment.tpl функция toggleEditForm. Там идёт прямая запись в outerHTML что уже само по себе хреново, было бы неплохо переписать нормально.

> Не работает (в веб-консоли "TypeError: this is not a function")

Насколько я понял строчка 364 там же. Вместо
`ls.comments.toggle this this.parentNode.dataset.id)`
должно быть 
`ls.comments.toggle this.parentNode.dataset.id)`

**Comment by zluca** [comment]
@Farxial 
> Насколько я понял

А нет, не так я понял, там 1й параметр нужен но видимо не this.

**Comment by zluca** [comment]
Ну эт достаточно редкая операция, к тому же только для админов. По логике построения интерфейсов то что не используется часто засовывается куда подальше.

**Comment by zluca** [comment]
@Farxial 
По идее если закомментировать строчку `obj.text result.sTextToggle` должно заработать, понять бы ещё что там за obj подразумевается.

**Comment by Farxial** [comment]
Что-то я ступил и сразу не догадался посмотреть в веб-консоли место в JS v.v
https://github.com/everypony/tabun/blob/development/frontend/app/comments.coffee#L365
Вроде, в CoffeScript `a b` означает `a(b)` и по такой логике `a b c` должно означать `a(b(c))`. Хотя могу и ошибаться, но, вроде бы, всё сходится.

**Comment by zluca** [comment]
Да, походу запятая пропущена. Вот так по идее: `ls.comments.toggle this, this.parentNode.dataset.id)`.

**Comment by Farxial** [comment]
Теперь комментарий помечается удалённым и есть уведомление об этом, но форма редактирования не пропадает. Хотя её можно закрыть вручную или открыть и закрыть сразу, а нажать "Удалить" потом. Но "Удалить" не заменяется на "Восстановить".

**Comment by zluca** [comment]
А кнопки "Восстановить" вообще нету. =)
Посмотри в toggleEditForm там надо добавить, и в стилях прописать чтобы она отображалась только внутри .comment-deleted, а "Удалить" наоборот не отображалась.
А ещё обработчик событий навесить. =)

**Comment by KarelWintersky** [comment]
На оригинальном ливстрите удаленный комментарий может восстановить админ. При этом в базе пишется в `is_deleted` 1. 

Если на табуне восстановить комментарий нельзя от слова нельзя совсем - то я предлагаю реализовать замену оригинального текста комментария  на блок:

`Комментарий от %%username%% удалён модератором %%censor%%`
